internal/extproc/translator/openai_gcpanthropic.go

```diff
@@ -6,59 +6,715 @@
+const (
+	anthropicVersionKey   = "anthropic_version"
+	gcpBackendError       = "GCPBackendError"
+	defaultMaxTokens      = int64(100)
+	tempNotSupportedError = "temperature %.2f is not supported by Anthropic (must be between 0.0 and 1.0)"
+)
+
+var errStreamingNotSupported = errors.New("streaming is not yet supported for GCP Anthropic translation")
+
+// openAIToGCPAnthropicTranslatorV1ChatCompletion where we can store information for streaming requests.
+type openAIToGCPAnthropicTranslatorV1ChatCompletion struct{}
+
+// Option defines a function that configures the translator.
+type Option func(*openAIToGCPAnthropicTranslatorV1ChatCompletion)
+
+// AnthropicContent Anthropic request/response structs.
+type AnthropicContent struct {
+	Type   string                            `json:"type"`
+	Text   string                            `json:"text"`
+	Source *anthropic.Base64ImageSourceParam `json:"source,omitempty"`
+}
+
+// NewChatCompletionOpenAIToGCPAnthropicTranslator creates a new translator.
 func NewChatCompletionOpenAIToGCPAnthropicTranslator() OpenAIChatCompletionTranslator {
 	return &openAIToGCPAnthropicTranslatorV1ChatCompletion{}
 }
 
-type openAIToGCPAnthropicTranslatorV1ChatCompletion struct{}
+func anthropicToOpenAIFinishReason(stopReason anthropic.StopReason) (openai.ChatCompletionChoicesFinishReason, error) {
+	switch stopReason {
+	// The most common stop reason. Indicates Claude finished its response naturally.
+	// or Claude encountered one of your custom stop sequences.
+	// TODO: A better way to return pause_turn
+	// TODO: "pause_turn" Used with server tools like web search when Claude needs to pause a long-running operation.
+	case anthropic.StopReasonEndTurn, anthropic.StopReasonStopSequence, anthropic.StopReasonPauseTurn:
+		return openai.ChatCompletionChoicesFinishReasonStop, nil
+	case anthropic.StopReasonMaxTokens: // Claude stopped because it reached the max_tokens limit specified in your request.
+		// TODO: do we want to return an error? see: https://docs.anthropic.com/en/docs/agents-and-tools/tool-use/implement-tool-use#handling-the-max-tokens-stop-reason
+		return openai.ChatCompletionChoicesFinishReasonLength, nil
+	case anthropic.StopReasonToolUse:
+		return openai.ChatCompletionChoicesFinishReasonToolCalls, nil
+	case anthropic.StopReasonRefusal:
+		return openai.ChatCompletionChoicesFinishReasonContentFilter, nil
+	default:
+		return "", fmt.Errorf("received invalid stop reason %v", stopReason)
+	}
+}
+
+// validateTemperatureForAnthropic checks if the temperature is within Anthropic's supported range (0.0 to 1.0).
+// Returns an error if the value is greater than 1.0.
+func validateTemperatureForAnthropic(temp *float64) error {
+	if temp != nil && (*temp < 0.0 || *temp > 1.0) {
+		return fmt.Errorf(tempNotSupportedError, *temp)
+	}
+	return nil
+}
+
+func isDataURI(uri string) bool {
+	return strings.HasPrefix(uri, "data:")
```

Comment:
nit; can you move this inline as it's only used in once place + not for unit-testability where you also have `strings.HasSuffix(strings.ToLower(imageURL), ".pdf"):` so let's make it consistent

Reply:
removed